### ISSUE TITLE
[camera_android] 0.10.9+5 breaks the preview's orientation

### ISSUE THREAD

**Issue by AlexV525** [flutter/flutter#150549]
### Steps to reproduce

1. `camera_android: 0.10.9+5`
2. Runs on a real device.
3. Choose the back lens.
4. The preview got rotated unexpectedly.

Roll back to v0.10.9+3 works as expected.

### Expected results

The preview should follow its actual orientation (portrait up).

### Actual results

The preview is rotated to landscape left.

### Code sample

https://pub.dev/packages/camera_android/example

### Screenshots or Video

| 0.10.9+3 ✅ | 0.10.9+5 ❌ |
|-------|-------|
| <img width="562" alt="image" src="https://github.com/flutter/flutter/assets/15884415/7942d4ef-4448-47bc-b95f-0fa8d5808caf"> | <img width="562" alt="image" src="https://github.com/flutter/flutter/assets/15884415/3ab6c833-d542-4c66-955c-06c36181e866"> |

### Logs

N/A

### Flutter Doctor output

```console
[✓] Flutter (Channel stable, 3.22.2, on macOS 14.5 23F79 darwin-arm64 (Rosetta), locale zh-Hans-CN)
    • Flutter version 3.22.2 on channel stable at /Users/alex/dev/flutter-322
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 761747bfc5 (2 weeks ago), 2024-06-05 22:15:13 +0200
    • Engine revision edd8546116
    • Dart version 3.4.3
    • DevTools version 2.34.3
```

**Comment by AlexV525** [comment]
cc @camsim99 

#149294 is related to targeting with another implementation package.

**Comment by AlexV525** [comment]
Migrating to the camerax implementation is unacceptable to downstream packages since:
1. The `camera_android` is not deprecated.
2. `flutter pub upgrade` will automatically upgrade the package to this broken version and shipped to the user end.
3. Both packages are still experimental (no major version releases yet).

**Comment by danagbemava-nc** [comment]
Reproducible using the plugin example app. This indeed works as expected with camera_android: 0.10.9+3 but does not work with 0.10.9+5

| 0.10.9+5 | 0.10.9+3 | 
| -- | -- |
| <img src="https://github.com/flutter/flutter/assets/88313112/a226a17f-dcce-417e-8477-569a477c1f02" width="300"/>| <img src="https://github.com/flutter/flutter/assets/88313112/47bb50e1-f2eb-4471-b080-603bddfe2b13" width="300"/> |


<details>
<summary>flutter doctor -v</summary>

```
[√] Flutter (Channel stable, 3.22.2, on Microsoft Windows [Version 10.0.22631.3155], locale en-US)
    • Flutter version 3.22.2 on channel stable at C:\Users\Deana\fvm\versions\stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 761747bfc5 (2 weeks ago), 2024-06-05 22:15:13 +0200
    • Engine revision edd8546116
    • Dart version 3.4.3
    • DevTools version 2.34.3

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\Deana\AppData\Local\Android\sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-b2043.56-10550314)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.8.2)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.8.34322.80
    • Windows 10 SDK version 10.0.22621.0

[√] Android Studio (version 2023.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-b2043.56-10550314)

[√] VS Code (version 1.87.2)
    • VS Code at C:\Users\Deana\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.90.0

[√] Connected device (4 available)
    • Pixel 7 (mobile)  • 28291FDH2001SA • android-arm64  • Android 14 (API 34)
    • Windows (desktop) • windows        • windows-x64    • Microsoft Windows [Version 10.0.22631.3155]
    • Chrome (web)      • chrome         • web-javascript • Google Chrome 126.0.6478.62
    • Edge (web)        • edge           • web-javascript • Microsoft Edge 124.0.2478.97

[√] Network resources
    • All expected network resources are available.

• No issues found!
```

```
[!] Flutter (Channel master, 3.23.0-13.0.pre.242, on Microsoft Windows [Version 10.0.22631.3155], locale en-US)
    • Flutter version 3.23.0-13.0.pre.242 on channel master at C:\Users\Deana\fvm\versions\master
    ! Warning: `flutter` on your path resolves to C:\Users\Deana\fvm\versions\stable\bin\flutter, which is not inside your current Flutter SDK checkout at C:\Users\Deana\fvm\versions\master. Consider adding C:\Users\Deana\fvm\versions\master\bin to the front of your path.
    ! Warning: `dart` on your path resolves to C:\Users\Deana\fvm\versions\stable\bin\dart, which is not inside your current Flutter SDK checkout at C:\Users\Deana\fvm\versions\master. Consider adding C:\Users\Deana\fvm\versions\master\bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5187cab7bd (5 days ago), 2024-06-14 23:35:34 -0400
    • Engine revision 9779c273aa
    • Dart version 3.5.0 (build 3.5.0-266.0.dev)
    • DevTools version 2.36.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\Deana\AppData\Local\Android\sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-b2043.56-10550314)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.8.2)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.8.34322.80
    • Windows 10 SDK version 10.0.22621.0

[√] Android Studio (version 2023.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-b2043.56-10550314)

[√] VS Code (version 1.87.2)
    • VS Code at C:\Users\Deana\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.90.0

[√] Connected device (4 available)
    • Pixel 7 (mobile)  • 28291FDH2001SA • android-arm64  • Android 14 (API 34)
    • Windows (desktop) • windows        • windows-x64    • Microsoft Windows [Version 10.0.22631.3155]
    • Chrome (web)      • chrome         • web-javascript • Google Chrome 126.0.6478.62
    • Edge (web)        • edge           • web-javascript • Microsoft Edge 124.0.2478.97

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

**Comment by AlexV525** [comment]
To be clear, only Flutter 3.22 can reproduce this because the SDK constraint has been set in the latest version of the plugin, so previous Flutter users won't resolve this version.

**Comment by camsim99** [comment]
This is a duplicate of https://github.com/flutter/flutter/issues/149294 I believe.

**Comment by AlexV525** [comment]
> This is a duplicate of #149294 I believe.

@camsim99 As I mentioned, #149294 and all its references are targeting the camera implementation (like PRs are only changing files with camerax), not the camera2, so I believe it's not a duplicate but only shares the same behavior.

**Comment by camsim99** [comment]
@AlexV525 Gotcha. Looks like camera_android may need a similar fix to the one camera_android_camerax is getting to support Impeller.

**Comment by camsim99** [comment]
FYI @jonahwilliams. Going to approve the revert.

**Comment by AlexV525** [comment]
Then we should continue tracking this with a further reland if I get this correct.

**Comment by yaakovschectman** [comment]
From triage: We suspect this is a duplicate of #149177

**Comment by AlexV525** [comment]
> From triage: We suspect this is a duplicate of #149177

Again, the issue is about the Camera2 implementation not the CameraX.

**Comment by reidbaker** [comment]
Fixed in 0.10.9+6
